ui/src/Pages/PhotosPage/PhotosPage.js

```diff
@@ -43,13 +55,59 @@ class PhotosPage extends Component {
       activeAlbumIndex: -1,
       activePhotoIndex: -1,
       presenting: false,
+      onlyWithFavorites: this.props.match.params.subPage === 'favorites',
     }
 
     this.setPresenting = this.setPresenting.bind(this)
     this.nextImage = this.nextImage.bind(this)
     this.previousImage = this.previousImage.bind(this)
 
     this.albums = []
+    this.refetchNeededFavorites = false
+    this.refetchNeededAll = false
+  }
+
+  onPopState(event) {
+    this.state.setState({
+      onlyWithFavorites: event.state.showFavorites,
+    })
+  }
+
+  componentDidMount() {
+    window.addEventListener('popstate', this.onPopState)
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('popstate', this.onPopState)
+  }
+
+  favoritesCheckboxClick(refetch) {
+    const onlyWithFavorites = !this.state.onlyWithFavorites
+    history.pushState(
```

Comment:
Should use `history.replaceState`

Reply:
Oh, I forgot this. You told me to change it to replaceState and I did it on Album's page but forgot this one.